src/services/aws_macros.ts

```diff
@@ -81,6 +81,16 @@ export class AWS {
   constructor(config: AWSConfig) {
     // declare an specific slow retry strategy, to reuse in slow apis
     this.slowRetryStrategy = new StandardRetryStrategy(async () => SLOW_STRATEGY_RETRIES, {
+      retryDecider: error => {
+        // copied the default behavior from aws-sdk to fix the problem caused by Jest on isRetryableByTrait
+        if (!error) {
+          return false;
+        }
+        if (error.name !== 'Error')
```

Comment:
but isn't jest overriding the `error` and not having the `name` property? If no `error.name` it will still go through this path?

Reply:
Jest is making the error not an instance of `Error`. Therefore the AWS SDK replaces it with a `new Error(...)` which has `name = 'Error'`.

Reply:
ohh got it!